terraform/main.tf

```diff
@@ -155,11 +155,11 @@ data "google_project" "project" {
 data "google_client_config" "default" {}
 
 module "cross-device" {
-  source                      = "./cross-device"
-  count                       = var.cross-device ? 1 : 0
-  project_id                  = data.google_project.project.id
-  region                      = var.region
-  spanner_instance_config     = var.spanner_instance_config
-  spanner_processing_units    = var.spanner_processing_units
-  list_nodepool_sa_iam_emails = local.list_nodepool_sa_iam_emails
+  count                    = var.cross-device ? 1 : 0
+  source                   = "./cross-device"
+  project_id               = data.google_project.project.id
+  region                   = var.region
+  spanner_instance_config  = var.spanner_instance_config
+  spanner_processing_units = var.spanner_processing_units
+  tenant_apps_sa_name      = format("%s-%s-apps-sa", var.cluster_name, "main")
```

Comment:
You're duplicating what we configured in the [tenant_apps_sa_name local variable](https://github.com/GoogleCloudPlatform/federated-learning/blob/main/terraform/main.tf#L30).

Also, you're defaulting to the service account that we use for Pods in the `main` namespace. Is this a deliberate design decision? If so, it should be documented.

What you need to do is:

1.  Introduce a new variable to let the user select the namespace where they are going to deploy workloads.
2. Get the related `tenant_apps_sa_email` by looking up that value in `module.service_accounts`. [Example](https://github.com/GoogleCloudPlatform/federated-learning/blob/main/terraform/main.tf#L65)